util/pkg/vfs/s3fs.go

```diff
@@ -155,6 +163,50 @@ func (p *S3Path) WriteFile(data io.ReadSeeker, aclObj ACL) error {
 	return nil
 }
 
+func (p *S3Path) GetVersioning() (bool, error) {
+	client, err := p.client()
+	if err != nil {
+		return false, err
+	}
+
+	request := &s3.GetBucketVersioningInput{}
+	request.SetBucket(p.bucket)
+
+	response, err := client.GetBucketVersioning(request)
+	if err != nil {
+		return false, fmt.Errorf("error getting s3 bucket versioning: %v", p, err)
+	}
+
+	if strings.Contains(response.String(),"Status: \"Enabled\"") {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func (p *S3Path) SetVersioning(status string) ( error) {
+	client, err := p.client()
+	if err != nil {
+		return err
+	}
+
+	requestConfig := &s3.VersioningConfiguration{}
```

Comment:
Please construct as an inline struct.  You can inline all of the struct members.

```go
requestConfig := &s3.VersioningConfiguration{
  SetStatus: status,
}
```

Reply:
It doesn't work for me somehow. It says "SetStatus not found"